Lesson4/step2/project.py

```diff
@@ -1,26 +1,29 @@
-from flask import Flask, render_template, request, redirect, jsonify, url_for, flash
-from sqlalchemy import create_engine, asc
-from sqlalchemy.orm import sessionmaker
-from database_setup import Base, Restaurant, MenuItem, User
-from flask import session as login_session
+import json
 import random
 import string
-from oauth2client.client import flow_from_clientsecrets
-from oauth2client.client import FlowExchangeError
+
 import httplib2
-import json
-from flask import make_response
 import requests
+from flask import Flask, render_template, request, \
+    redirect, jsonify, url_for, flash, make_response, \
+    session as login_session
+from oauth2client.client import FlowExchangeError, OAuth2WebServerFlow
+from sqlalchemy import asc
+from sqlalchemy.orm import sessionmaker
+
+from database_setup import Base, Restaurant, MenuItem, User, engine
+from variables import facebook_web, google_web, main_app
+
+state = ''.join(random.choice(string.ascii_uppercase + string.digits)
+                for x in range(32))
 
 app = Flask(__name__)
+app.secret_key = state
 
-CLIENT_ID = json.loads(
-    open('client_secrets.json', 'r').read())['web']['client_id']
+GOOGLE_CLIENT_ID = google_web.get('client_id')
 APPLICATION_NAME = "Restaurant Menu Application"
 
-
 # Connect to Database and create database session
-engine = create_engine('sqlite:///restaurantmenuwithusers.db')
```

Comment:
Pull in the engine and other information from a central location.  This can probably further refactored since the same calls get made to other parts.